### PR TITLE
script to prepare remote cluster

### PR DESCRIPTION
**Pull request by jmazzitelli**
fixes: https://github.com/kiali/kiali/issues/5987

This renames the existing hack script and provides a way to create the remote clusterrole/SA without also requiring the user to create a secret in the kiali namespace.
This allows the user to prepare the remote cluster and they can create the remote secret later.

### CODE REVIEW

**Comment by jmazzitelli** [comment]
I tested with one OpenShift cluster (context=current) and one Minikube cluster (context=ci).

1. You can do a dry run and see what resources will be created:
```
hack/istio/multicluster/kiali-prepare-remote-cluster.sh -rcc ci -kcc current --dry-run true 2>/dev/null
```
results in all the resources including the remote cluster secret that has this in it (this is the secret that Kiali will use to obtain the information necessary to connect to the remote cluster):
```yaml
    clusters:
    - name: ci
      cluster:
        server: https://192.168.59.254:8443
        certificate-authority-data: LS0tLS1CRUdJT...0tCg==
```

```
hack/istio/multicluster/kiali-prepare-remote-cluster.sh -rcc current -kcc ci --dry-run true --allow-skip-tls-verify true 2>/dev/null
```
results in all the resources including the remote cluster secret that has this in it (this is the secret that Kiali will use to obtain the information necessary to connect to the remote cluster):
```yaml
    clusters:
    - name: api-ocp4-local:6443
      cluster:
        server: https://api.ocp4.local:6443
        insecure-skip-tls-verify: true
```

The above was using `--dry-run true`. Now tell the script to create the resources.

```
hack/istio/multicluster/kiali-prepare-remote-cluster.sh -rcc current -kcc ci -prr true -pks false
```

This will create the service account and related resources, but will skip creating the kiali secret. The end of the output looks like:
```
INFO: Create the remote cluster namespace [kiali-access-ns] if it doesn't exist
namespace/kiali-access-ns created
INFO: Create the remote cluster resources with the appropriate permissions for Kiali (view_only=true)
serviceaccount/kiali-remote-access created
clusterrole.rbac.authorization.k8s.io/kiali-remote-access-viewer created
clusterrolebinding.rbac.authorization.k8s.io/kiali-remote-access created
secret/kiali-remote-access created
INFO: Skipping the creation of the Kiali remote cluster secret.
```

Now ask the script to create the kiali secret:
```
hack/istio/multicluster/kiali-prepare-remote-cluster.sh -rcc current -kcc ci -prr false -pks true --allow-skip-tls-verify true
```

Results in:
```
INFO: Skipping the creation of the remote resources.
INFO: Create the remote cluster secret in the Kiali cluster
INFO: remote_cluster_server_url=https://api.ocp4.local:6443
INFO: Kiali will be allowed to insecurely skip TLS verification when connecting to the remote cluster named [api-ocp4-local:6443].
The Secret "kiali-remote-cluster-secret-api-ocp4-local-6443" is invalid: data[api-ocp4-local:6443]: Invalid value: "api-ocp4-local:6443": a valid config key must consist of alphanumeric characters, '-', '_' or '.' (e.g. 'key.name',  or 'KEY_NAME',  or 'key-name', regex used for validation is '[-._a-zA-Z0-9]+')
```

That's a bug in the script that I need to fix. Stay tuned for update.

**Review by leandroberetta — COMMENTED** [review]
Looks good. I tried the primary-remote script and works fine. 

Just a comment about a commented line in the primary-remote script that we should remove it now (it was introduced in a previous PR).

File: hack/istio/multicluster/install-primary-remote.sh [hunk 1]
```diff
@@ -38,7 +38,7 @@ EXPOSE_SERVICES_YAML="${ISTIO_DIR}/samples/multicluster/expose-services.yaml"
 ${CLIENT_EXE} apply --context=${CLUSTER1_CONTEXT} -n ${ISTIO_NAMESPACE} -f $EXPOSE_SERVICES_YAML
 
 # Create Kiali remote secrets
-#source ${SCRIPT_DIR}/kiali-create-remote-cluster-secret.sh
+#source ${SCRIPT_DIR}/kiali-prepare-remote-cluster.sh
```
[the remark this answers is not in the document]
Comment (jmazzitelli) [reply hack/istio/multicluster/install-primary-remote.sh]: done in latest commit

**Review by leandroberetta — APPROVED** [review]
Looks good. I read the help and it's clear. I also tested the primary-remote script and works as expected, the secrets are created.

**Pull request merged**